### PR TITLE
[10.0][IMP] ir_attachment modules

### PR DESCRIPTION
**Pull request by rbikbov**

### CODE REVIEW

File: ir_attachment_s3/doc/index.rst [hunk 1]
```diff
@@ -65,6 +65,16 @@ Configuration
 
 The settings are also available from the ``Settings >> Technical >> Database Structure >> S3 Settings``.
 
+* You can also set these parameters using environment variables
+
+  Mapping Environment variables to Odoo System parameters:
+  * ``S3_BUCKET``: it is ``s3.bucket`` parameter
```
Comment (yelizariev) [inline ir_attachment_s3/doc/index.rst:72]: add extra empty line

image

File: ir_attachment_s3/doc/index.rst [hunk 2]
```diff
@@ -65,6 +65,16 @@ Configuration
...
+  * ``S3_CONDITION``: it is ``s3.condition`` parameter
+  * ``S3_ACCESS_KEY_ID``: it is ``s3.access_key_id`` parameter
+  * ``S3_SECRET_KEY``: it is ``s3.secret_key`` parameter
+
+  NOTE: if you simultaneously set parameters both in the Environment variables and in the Odoo System parameters, then the Odoo System parameters have the highest priority.
```
Comment (yelizariev) [inline ir_attachment_s3/doc/index.rst:77]: :+1:

**Review by yelizariev — CHANGES_REQUESTED** [review]

**Comment by yelizariev** [comment]
There is no :fix: emoji :cry:

File: ir_attachment_s3/models/ir_attachment.py [hunk 3]
```diff
@@ -37,9 +37,6 @@ def _get_s3_settings(self, param_name, os_var_name):
         res = config_obj.sudo().get_param(param_name)
         if not res:
             res = os.environ.get(os_var_name)
-            if res:
-                config_obj.sudo().set_param(param_name, res)
-                _logger.info('parameter {} has been created from env {}'.format(param_name, os_var_name))
```
Comment (yelizariev) [inline ir_attachment_s3/models/ir_attachment.py:42]: Update changelog for this

File: ir_attachment_s3/doc/index.rst (hunk 1 above)
Comment (rbikbov) [reply ir_attachment_s3/doc/index.rst:72]: Done.

File: ir_attachment_s3/models/ir_attachment.py (hunk 3 above)
Comment (rbikbov) [reply ir_attachment_s3/models/ir_attachment.py:42]: Done.

**Review by yelizariev — APPROVED** [review]
Code is fine

**Comment by yelizariev** [comment]
Tests are not passed

**Comment by yelizariev** [comment]
When it's ready you have to cleanup commits and use release tags only

File: .travis.yml [hunk 4]
```diff
@@ -18,6 +18,9 @@ env:
...
+  - secure: "IFd+iJjl2lUql3Bc7MVWk+0J5wz977v6KYXolh6APbiDoBJ5MNtr8leDe1thf3LjKM590Hq1/Q05COaJVYxo/6+GZh90un/8MGTuQ/nOAbyozhHWPj5MHeOkDWowlJxlNRqNXttpT9EuhJpRJbC9uXUXeFnf1EYAluyB+V3M4SX94kgGEVp71kp7E20lcWOR4oGslYBO4uvpPudtG8xA7jaXVSvpYuBbX07SzdPvniIVjJ6mLGSmnxGkcjJJmLXyUSyVFlJYbXtrLu2t1iechyQBhZpyyHD0k+0vaf/0b4AAzR9oyosJUZ5j9RmxX+mYEldSlayEkrfD8BDHdeqzHvXJe0VCo9w80NxsHEDUbwPIZp/D7tofmNjL3hq67/HjSSjQrI8XBjzrvUwG8xT4pf0yzFCvRvpzQPbNfNLDYAM4o1X016TjtuLjI4+w4UxcSVzjF5LVfBotSL+ok1GtllYy40Bf/76EFrrQoy7EmSDZaID1aUx0DptdlX4CsKb6inTIDIXcAH2AyMtLjAVfxfEuV5X1UJRsxGvpkIvunPtpcyds9DBBrUOEXSoZUyLCNbQGawfJt1hgT2eyYcxotXnF20Bcqy24qpNzJoBTsiE8VAeS8xRv3RVwMIgXDTkn0vX9U4XMNu7XtSg0eYQoMgimHJdAepbYseUwAf/B+x4="
```
Comment (yelizariev) [inline .travis.yml]: Note, that travis  uses it only in published branches, not in PR. 
Also, add a comment which those variables are about (S3 credentials?)

**Review by yelizariev — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 4 above)
Comment (rbikbov) [reply .travis.yml]: Done.

**Review by yelizariev — CHANGES_REQUESTED** [review]
Looks very professional, but one commit has wrong tag. Replace :fire:  to :shield:

**Comment by yelizariev** [comment]
@Ramil-Mukhametzyanov or @ilmir-k  (decide yourself who will check it)
updates affect products: at backend (variant and template) at ecommerce (/shop, product pages) -- everywhere it must be url to s3 file with proper size. It cannot be base64 data (check it via network or via "Open image in new tab")

**Review by Ramil-Mukhametzyanov — CHANGES_REQUESTED** [review]
There is no any record about ir_attachment_url changes in changelog and version is not changed.

**Comment by KolushovAlexandr** [comment]
superseded by https://github.com/it-projects-llc/misc-addons/pull/775

**Pull request closed without merging**